todo.txt

```diff
@@ -0,0 +1,38 @@
+do sprawdzenia:
+
+QUSTION 1. test stworzenia nowego Pull Requesta
+	jesli dobrze rozumiem, to PR moze istniec tylko jak ktos stworzy nowy branch i wysle jakis commit? 
+	A moze ktos moze: a) zrobic lokalnie zmiany w master, b) git push origin master ?
+	
+ANSWER 1 
+	- wchodzi na to, ze mozna z byle czego, ale topic branch jest lepsza opcja bo mozna "follow-ups" 
+	https://help.github.com/articles/about-pull-requests/
+	ale ogolnie sa sa dwa przypadki: 
+	a) contributor ma write access do projektu, wtedy napewno musi miec nowy branch (jak nie ma to przycisk "create pull request" nie jest aktywny),
+	b) constributor nie ma write access, to wtedy moze i da sie wrzucic z master do master jak uzywa sie fork'a? 
```

Comment:
Fork to jest kopia repozytorium, z zapisaną informacją skąd została skopiowana. Jeśli nie posiada Pan write-access do danego repozytorium a chciałby Pan w nim coś zmienić - forkuje się je na swoje konto. Nazwa projektu pozostaje bez zmian, ale zmienia się jego właściciel. Dlatego po sforkowaniu, zyskujemy identyczne repozytorium jak repozytorium źródłowe i możemy je od teraz traktować jak swoje - czyli pushować gdzie chcemy bo do swojego mamy pełny write access.

Różnica jest taka że commity który wpushujemy do brancha "master" czy do dowolnego innego w naszym forku nie pojawią się w repozytorium źródłowym. Żeby się pojawiły - należy stworzyć pull request (tylko taki "większy" międzyrepozytoriowy) w którym powiemy, że chcemy dołączyć do projektu źródłowego commity, które w naszej kopii (forku) repozytorium są nowe w stosunku do oryginalnego.

W zespole, jeśli wszyscy commitujący są dodani do repozytorium - tak jak Pan napisał - przypadek 1. Przypadek 2 stosujemy w nienaszych projektach, które chcemy jakoś wspomóc (np. znalazłem błąd gdzieś i chcę zgłosić poprawkę albo brakuje mi jakiejś funkcjonalności więc ją dopisuję i proponuję w postaci PR zamiast prosić jednego z autora projektów o jej napisanie). Na tym polega magia i siła otwartego kodu :-)

Reply:
tak, idee fork'a czuje dobrze, ale to co pan napisal 
"należy stworzyć pull request (tylko taki "większy" międzyrepozytoriowy) w którym powiemy, że chcemy dołączyć do projektu źródłowego commity, które w naszej kopii (forku) repozytorium są nowe w stosunku do oryginalnego."
To bede musial kiedys przecwiczyc. 
Zasadniczo widze nastepujace opcje:
- mam write access (moje repo lub organizacja) - wtedy jak puszcze commity do mastera, nie da sie z nich zrobic PR, PR w tym wypadku tylko z nowym branch'em
- nie mam write access (chodzby projekt kolegi na jego koncie github'owym) - musze zrobic fork'a, dodac swoje commity i tworzyc do niego PR? 
- w zwiazku z w/w rozumiem, ze np. Pan nie moze zrobic mi PR takiego normalnego? (bo nie ma Pan do mnie write access?), jedyna opcja jakby Pan zrobic fork i wtedy otworzyl PR?
correct?

Reply:
Dokładnie tak jak Pan napisał